src/main/java/io/fabric8/maven/docker/assembly/DockerFileBuilder.java

```diff
@@ -235,14 +237,48 @@ private String quote(String value) {
 
     private void addPorts(StringBuilder b) {
         if (ports.size() > 0) {
-            String[] portsS = new String[ports.size()];
-            int i = 0;
-            for (Integer port : ports) {
-                portsS[i++] = port.toString();
+            String[] portsS = ports.toArray(new String[]{});
+            for(String port : portsS) {
+            	validatePortMapping(port);
             }
             DockerFileKeyword.EXPOSE.addTo(b, portsS);
         }
     }
+    
+    // Pattern for splitting of the protocol part of the port.  
+    private static final Pattern PROTOCOL_SPLIT_PATTERN = Pattern.compile("(.*?)(?:/(tcp|udp))?$");
```

Comment:
Guess we can match case insensitive here, and do a lowercase when adding them to the port list. makes it a bit more flexible for users.